src/utils.jl

```diff
@@ -561,6 +563,27 @@ function batch(xs)
   return data
 end
 
+"""
+  unbatch(x)
+
+Reverse of the [`batch`](@ref) operation,
+unstacking the last dimension of the array `x`.
+
+See also [`unstack`](@ref).
+
+# Examples
+
+```jldoctest
+julia> Flux.unbatch([1 3 5 7; 
+                     2 4 6 8])
+4-element Vector{Vector{Int64}}:
+ [1, 2]
+ [3, 4]
+ [5, 6]
+ [7, 8]
+"""
+unbatch(x::AbstractArray) = unstack(x, ndims(x))
```

Comment:
`batch` has this weird implementation where if the vector is not a vector of arrays, then it is a no-op. I suppose if we want perfect symmetry, we need a dispatch for this?

```suggestion
unbatch(x::AbstractVector) = x
unbatch(x::AbstractArray) = unstack(x, ndims(x))
```

Maybe we can also clean up the `batch` implementation too.

```julia
batch(x::AbstractVector{<:AbstractArray}) = stack(x, ndims(first(x)) + 1)
batch(x::AbstractVector) = x
```